<launch>
    <!--
    <node name="left_wheel_pid" pkg="gazebo_ros" type="controller" output="screen" >
      <param name="Kp" value="5.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.1" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="20" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <param name="setpoint_timeout" value="-1.0" />
     </node>
    -->
    <rosparam file="$(find braccio_urdf_description)/launch/controller.yaml" command="load"/>
    <node args="giunto1_position_controller giunto2_position_controller giunto3_position_controller giunto4_position_controller giunto5_position_controller giunto0_position_controller joint_state_controller " name="controller_spawner" ns="braccio_urdf" output="screen" pkg="controller_manager" respawn="false" type="spawner"/>
    <node name="robot_state_publisher" output="screen" pkg="robot_state_publisher" respawn="false" type="robot_state_publisher">
        <remap from="/joint_states" to="/braccio_urdf/joint_states"/>
    </node>

    <!--
    <node name="servo_sim_node" pkg="gazebo_ros" type="plant_sim" output="screen" >
      <param name="plant_order" value="2" />
    </node>

    <node name="setpoint_node" pkg="gazebo_ros" type="setpoint_node" output="screen" />
    -->
    
    <!-- rqt_plot is a resource hog, so if you're seeing high CPU usage, don't launch rqt_plot  -->
    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/control_effort/data /state/data /setpoint/data" />

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

    <node name="rqt_robot_monitor" pkg="rqt_robot_monitor" type="rqt_robot_monitor" />
</launch>
